[Bug]: gotaglib WASM panic on malformed files crashes the scanner subprocess

### I confirm that:

- [x] I have searched the existing [open AND closed issues](https://github.com/navidrome/navidrome/issues?q=is%3Aissue) to see if an issue already exists for the bug I've encountered
- [x] I'm using the latest version (your issue may have been fixed already)

### Version

0.60.3

### Current Behavior

When the library scanner encounters an audio file with malformed or unusual tags, the gotaglib WASM extractor panics inside \`f.AllTags()\` or \`f.Properties()\`, crashing the entire scanner subprocess with \`panic: memory error\`.

The existing \`recover()\` in \`openFile()\` only catches panics during **file opening**, but the panic happens **after** \`openFile()\` returns, in \`extractMetadata()\`, so it is not caught. The panic propagates through the pipeline and terminates the subprocess.

### Expected Behavior

Files that cause a WASM memory panic should be logged and skipped gracefully, allowing the rest of the scan to continue. The scanner should not crash.

### Steps To Reproduce

1. Have an audio file with unusual or externally-written tags (e.g. lyrics embedded by a third-party tagger)
2. Trigger a library scan
3. Observer the scanner subprocess crash with `panic: memory error`

### Relevant log output

```
panic: memory error
goroutine 1216 [running]:
go.senan.xyz/taglib.readString(0xc000cec120, 0x617d0)
    taglib.go:1343 +0x265
go.senan.xyz/taglib.readStrings(...)
go.senan.xyz/taglib.(*wasmStrings).decode(...)
go.senan.xyz/taglib.(*module).call(...)
go.senan.xyz/taglib.(*File).RawTags(...)
go.senan.xyz/taglib.(*File).AllTags(...)
github.com/navidrome/navidrome/adapters/gotaglib.extractor.extractMetadata(...)
    gotaglib.go:58
...
level=error msg="Scan failed" error="failed to read status from scanner: ... exited with non-zero status code: exit status 2"
```

### Environment

- OS: Linux (Docker)
- Installed via: Docker

### Code of Conduct

- [x] I agree to follow Navidrome's Code of Conduct

--- a/adapters/gotaglib/gotaglib.go
+++ b/adapters/gotaglib/gotaglib.go
@@ -58,7 +58,19 @@ func (e extractor) Version() string {
 	return "unknown"
 }
 
-func (e extractor) extractMetadata(filePath string) (*metadata.Info, error) {
+func (e extractor) extractMetadata(filePath string) (info *metadata.Info, err error) {
+	// Recover from panics in the WASM runtime that can occur during any taglib
+	// operation (opening, reading tags, or reading properties). This catches crashes
+	// from malformed files or WASM runtime issues (e.g., wazero mmap failures on
+	// hardened systems with MemoryDenyWriteExecute=true).
+	debug.SetPanicOnFault(true)
+	defer func() {
+		if r := recover(); r != nil {
+			log.Error("gotaglib: WASM runtime panic reading file. Skipping", "file", filePath, "panic", r)
+			err = fmt.Errorf("WASM runtime panic: %v", r)
+		}
+	}()
+
 	f, close, err := e.openFile(filePath)
 	if err != nil {
 		log.Warn("gotaglib: Error reading metadata from file. Skipping", "filePath", filePath, err)
@@ -112,16 +124,6 @@ func (e extractor) extractMetadata(filePath string) (*metadata.Info, error) {
 // openFile opens the file at filePath using the extractor's filesystem.
 // It returns a TagLib File handle and a cleanup function to close resources.
 func (e extractor) openFile(filePath string) (f *taglib.File, closeFunc func(), err error) {
-	// Recover from panics in the WASM runtime (e.g., wazero failing to mmap executable memory
-	// on hardened systems like NixOS with MemoryDenyWriteExecute=true)
-	debug.SetPanicOnFault(true)
-	defer func() {
-		if r := recover(); r != nil {
-			log.Error("WASM runtime panic: This may be caused by a hardened system that blocks executable memory mapping.", "file", filePath, "panic", r)
-			err = fmt.Errorf("WASM runtime panic (hardened system?): %v", r)
-		}
-	}()
-
 	// Open the file from the filesystem
 	file, err := e.fs.Open(filePath)
 	if err != nil {

--- a/adapters/gotaglib/gotaglib.go
+++ b/adapters/gotaglib/gotaglib.go
@@ -67,6 +67,7 @@ func (e extractor) extractMetadata(filePath string) (info *metadata.Info, err er
 	defer func() {
 		if r := recover(); r != nil {
 			log.Error("gotaglib: WASM runtime panic reading file. Skipping", "filePath", filePath, "panic", r)
+			debug.PrintStack()
 			err = fmt.Errorf("WASM runtime panic: %v", r)
 		}
 	}()

--- a/adapters/gotaglib/gotaglib.go
+++ b/adapters/gotaglib/gotaglib.go
@@ -66,7 +66,7 @@ func (e extractor) extractMetadata(filePath string) (info *metadata.Info, err er
 	debug.SetPanicOnFault(true)
 	defer func() {
 		if r := recover(); r != nil {
-			log.Error("gotaglib: WASM runtime panic reading file. Skipping", "file", filePath, "panic", r)
+			log.Error("gotaglib: WASM runtime panic reading file. Skipping", "filePath", filePath, "panic", r)
 			err = fmt.Errorf("WASM runtime panic: %v", r)
 		}
 	}()

